### PR TITLE
[12.0] rename easy my coop to cooperator

### LINKED ISSUE #203: Rename `easy_my_coop` to `cooperator`
- [x] rename `easy_my_coop` to `cooperator`
- [x] merge  with property_cooperator_account functionality from `easy_my_coop_es`
- [x] repercussions 

related PR : 
- https://github.com/coopiteasy/vertical-cooperative/pull/202
- https://github.com/beescoop/Obeesdoo/pull/312
- https://github.com/coopiteasy/investor-wallet-platform/pull/74

### PR DESCRIPTION
**Pull request by robinkeunen**
# POC

- [x] rename easy_my_coop to cooperator and fix inheriting xml_ids
- [x] check install on a blank database
	- install cooperator
	- install easy_my_coop_website
	- install easy_my_coop_api
	- => install, form and unit test pass
- [x] migrate a database from easy_my_coop to cooperator
	- create db with emc installed + data generated
	- try pre install hooks w/ auto_install: does not work
	- pre update rename_deprecated_modules.py
	- run script then update all : 
		- data seems correct, 
		- basic flows pass
		- warning 1 remains, it is harmless but a little cleanup would be nice.
	- Error (2) was fixed in commit https://github.com/coopiteasy/vertical-cooperative/pull/202/commits/1a819c850f6189d5cf334e6f64950cc94dcd29d8. The reference to `easy_my_coop` remained on the report action since it was marked "noupdate=True"
	- note: works if ran multiple times
- [x] migrate a database from easy_my_coop and dependencies
  	- create db with emc, emc_api, emc_website installed + data generated
	- run script then update all : 
		- data seems correct, 
		- basic flows pass
		- warning 1 remains, it is harmless but a little cleanup would be nice.

(1) `2022-01-28 14:37:26,372 41226 WARNING emc odoo.modules.module: module easy_my_coop: module not found`

(2) `ValueError: External ID not found in the system: easy_my_coop.theme_invoice_G002`

# TODO

- [x] rename `easy_my_coop` to `cooperator`
- [x] fix dependencies in vertical-cooperative
- [x] fix dependencies in investor-wallet-platform
- [x] fix dependencies in obeesdoo
- [x] check the rest of the codebase
- [x] fix translations
- [x] ~~merge  with property_cooperator_account functionality from `easy_my_coop_es`~~\*

# TEST

- [x] install cooperator (plus other modules) on blank db & test
- [x] install iwp_website on blank db ~~& test~~ => tests are unfathomably broken for other reasons than this...
- [x] install easy_my_coop on blank db, run script and update to cooperator
- [x] run script and update oufticoop
- [x] run script and update synergie
- [x] run script and update lenid
- [x] run script and update brasseriedurenard 

 
\* the spanish l10n actually removes `("internal_type", "=", "receivable")` from the domain of `property_cooperator_account` rather than adding `("deprecated", "=", False)` (already in the domain).
=> leaving as is.

### CODE REVIEW

**Comment by carmenbianca** [comment]
TODO: Remove `@expectedFailure` from adjusted tests in 9102562

File: rename_deprecated_modules.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2021 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import logging
+
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+env = env  # noqa F821 undefined name 'env'
```
Comment (carmenbianca) [inline rename_deprecated_modules.py]: No need to edit this now, but this won't be parsed correctly. The correct syntax is `# noqa: F821`.

File: rename_deprecated_modules.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+renamed_modules = {
+    "easy_my_coop": "cooperator",
+}
+
+logging.info("rename easy_my_coop_x modules to cooperator_x")
```
Comment (carmenbianca) [inline rename_deprecated_modules.py]:
```suggestion
_logger.info("rename easy_my_coop_x modules to cooperator_x")
```

File: cooperator/readme/INSTALL.rst [hunk 3]
```diff
@@ -1,3 +1,11 @@
+## Migrating from easy_my_coop* to cooperator modules
+
+Run this command before updating your modules
+
+.. code-block:: shell
+
+  cat rename_deprecated_modules.py | ./odoo/odoo-bin shell -c odoo.conf --no-http  --stop-after-init -d <db-name>
+
```
Comment (robinkeunen) [inline cooperator/readme/INSTALL.rst:8]: Move to repository README.rst

**Pull request merged**